symphony/content/content.systemauthors.php

```diff
@@ -625,6 +627,7 @@ public function __actionNew()
                      */
                     Symphony::ExtensionManager()->notifyMembers('AuthorPostCreate', '/system/authors/', array(
                         'author' => $this->_Author,
+                        'author_id' => $author_id,
```

Comment:
Is this still required with the changes to `Author::commit`? `$this->_Author` will now include the ID.

Reply:
No it is not.

This is only to make it possible to reuse the same code as other delegates. I often find myself reusing the same function for multiple delegates and not having to look into the $context is a good win.

Reply:
On the other hand, doesn't a 3.0 release allow to clean stuff up? So what about removing the ID from all delegates? :-)

Reply:
Ah consistency makes sense!

Reply:
@michael-e I thought about it, and it felt weird when deleting the author: do we really want to load an Author object from the DB before deleting it ? If so, I would not mind removing the 'author_id'!

Reply:
Talking about consistency… :-)  …take a look at `content.publish.php`, for example (and there are more examples, IMHO). There you will find the same old scheme: create and edit delegates provide the object, but the delete delegate provides the ID only. 

So changing the author class only is actually bad if we talk about consistency. Shoudn't we have comparable delegates for authors, sections, entries etc.?

Reply:
> Shouldn't we have comparable delegates for authors, sections, entries etc.?

I think we should. Since they were all added in a "ad-hoc" fashion, simply filling the goal of the original developer.